lib/tom_queue/worker_supervisor.rb

```diff
@@ -0,0 +1,67 @@
+module TomQueue
+# class WorkerPool?
+  class WorkerSupervisor
+
+    attr_reader :processes
+
+    def initialize(processes = default_processes)
+      @processes = processes
+      setup_signal_handling
+    end
+
+    def default_processes
+      # TODO: deferred worker
+      {
+        "worker1" => -> { TomQueue::Worker.new.start },
+        "worker2" => -> { TomQueue::Worker.new.start },
+        "worker3" => -> { TomQueue::Worker.new.start },
+        "worker4" => -> { TomQueue::Worker.new.start },
+        "scheduler" => -> { TomQueue::Worker.new.start }
+      }
+    end
+
+    def setup_signal_handling
+      ["SIGINT","SIGTERM"].each do |signal|
+        Signal.trap(signal) do
+          raise SignalException, "SIGTERM"
+        end
+      end
+    end
+
+    def start_workers
+      pids = {}
+      # TODO: before_fork
```

Comment:
Again - I'd be inclined to offer this (before and after fork hooks) as an external method that can be set by the caller.

so the worker script would look something like:
```
supervisor = TomQueue::Supervisor.new
supervisor.supervise(as: "worker", count: 4) { TomQueue::Worker.new.start } 
supervisor.supervise(as: "deferrals") { TomQueue::DeferralManager.start }
supervisor.before_fork { Rails.application.config.before_fork.call() }
supervisor.after_fork { Rails.application.config.after_fork() }
supervisor.run
```

Does that make sense ?